src/utils.js

```diff
@@ -58,6 +58,36 @@ export function getRootExpression(node) {
 }
 
 /*
+ * returns the name of the reference node passed.
+ *
+ * example: var foo = document;
+ *  The node for foo will return 'document'
+ */
+export function getNodeReferenceName(context, node) {
+  var variables = context.getScope().variables;
+  var scopeVar;
+
+  // Finds variable reference in current scope
+  for (let variable of variables) {
+    if (variable.name === node.name) {
+      scopeVar = variable;
+      break;
+    }
+  }
+
+  if (scopeVar && scopeVar.defs && scopeVar.defs[0] &&
+      scopeVar.defs[0].node && scopeVar.defs[0].node.init &&
+      scopeVar.defs[0].node.init !== null) {
+    // Return the name of the first definition of the variable which
+    // corresponds to the node passed in.
```

Comment:
So what would happen if you re-defined the var? Obviously there's limits to what lengths we need to go to here - but, I feel it's important to understand what those limits are and document them so it's clear.